src/lavinmq/clustering/controller.cr

```diff
@@ -1,25 +1,31 @@
-require "../launcher"
 require "../etcd"
 require "./client"
 
 class LavinMQ::Clustering::Controller
   Log = LavinMQ::Log.for "clustering.controller"
 
-  @id : Int32
+  getter id : Int32
 
-  def initialize(@config = Config.instance)
-    @etcd = Etcd.new(@config.clustering_etcd_endpoints)
+  @repli_client : Client? = nil
+
+  def self.new(config : Config)
+    etcd = Etcd.new(config.clustering_etcd_endpoints)
+    new(config, etcd)
+  end
+
+  def initialize(@config : Config, @etcd : Etcd)
     @id = clustering_id
     @advertised_uri = @config.clustering_advertised_uri ||
                       "tcp://#{System.hostname}:#{@config.clustering_port}"
   end
 
-  def run
+  def run(&)
```

Comment:
Maybe document what the yield is for, and that the method will block after yielding